tests/test_middleware.py

```diff
@@ -1035,6 +1029,9 @@ class FaultyComponentC:
         def process_response_async(self, req, resp, resource, req_succeeded):
             pass
 
+    if util.HAS_CYTHON:
+        pytest.skip(reason='Cythonized coroutine functions cannot be detected')
```

Comment:
is this still true in cython>3?

Reply:
I was trying to focus on porting from `_util`, we can take a look at this separately. But probably it is easier to skip, because detection requires not only Cython 3.x, but also CPython 3.10+.

Reply:
ok, may make sense to understand if we still need to do this, since it may lead to simplification in the code too